core/common/src/main/java/alluxio/underfs/MultiRangeObjectInputStream.java

```diff
@@ -126,6 +129,7 @@ private long getBlockSize() {
   /**
    * Opens a new stream at mPos if the wrapped stream mStream is null.
    */
+  @EnsuresNonNull("mStream")
```

Comment:
What's the rule of thumb of using this rule?

Reply:
That's is basically saying after executing the annotated method sucessfully, we can guarantee the member variable ```mStream``` will not be null:
https://checkerframework.org/api/org/checkerframework/checker/nullness/qual/EnsuresNonNull.html

I will double check this code, I want to stick to only use ```@Nullable``` annotations, that way Alluxio developers don't need to learn too many Checker Framework annotations.

Reply:
I see. I feel the contract is that any methods without `@Nullable` annotation can be assumed to return non-null values.

Reply:
@yupeng9  Yes, you are right, but in this case, the method ```openStream``` does not return a value, instead changes the state of the member variable ```mStream```. 

For instance, consider the following method:
```
  @Override
  public int read(byte[] b, int offset, int length) throws IOException {
    if (length == 0) {
      return 0;
    }
    openStream();
    int read = mStream.read(b, offset, length);
    if (read != -1) {
      mPos += read;
      closeStreamIfBoundary();
    }
    return read;
  }
```

Because the ```openStream``` method is annotated with ```@EnsuresNonNull("mStream")```, the call to ```int read = mStream.read(b, offset, length)``` will never have a NPE, since we ensure that after calling the method ```openStream``` the member variable ```mStream``` will never be null.